src/libguac/guacamole/private/mem.h

```diff
@@ -0,0 +1,317 @@
+ *     such a block could not be allocated. If a block of memory could not be
+ *     allocated, guac_error is set appropriately.
+ */
+void* PRIV_guac_mem_alloc(size_t factor_count, const size_t* factors);
+
+/**
+ * Allocates a contiguous block of memory with the specified size and with all
+ * bytes initialized to zero, returning a pointer to the first byte of that
+ * block of memory. If multiple sizes are provided, these sizes are multiplied
+ * together to produce the final size of the new block. If memory of the
+ * specified size cannot be allocated, or if multiplying the sizes would result
+ * in integer overflow, guac_error is set appropriately and NULL is returned.
+ *
+ * This function is analogous to the standard calloc(), but accepts a list of
+ * size factors instead of a requiring exactly two integer sizes.
+ *
+ * The pointer returned by PRIV_guac_mem_zalloc() SHOULD be freed with a
+ * subsequent call to guac_mem_free() or PRIV_guac_mem_free(), but MAY instead
+ * be freed with a subsequent call to free().
+ *
+ * @param factor_count
+ *     The number of factors to multiply together to produce the desired block
+ *     size.
+ *
+ * @param factors
+ *     An array of one or more size_t values that should be multiplied together
+ *     to produce the desired block size. At least one value MUST be provided.
+ *
+ * @returns
+ *     A pointer to the first byte of the allocated block of memory, or NULL if
+ *     such a block could not be allocated. If a block of memory could not be
+ *     allocated, guac_error is set appropriately.
+ */
+void* PRIV_guac_mem_zalloc(size_t factor_count, const size_t* factors);
+
+/**
+ * Multiplies together each of the given values, storing the result in a size_t
+ * variable via the provided pointer. If the result of the multiplication
+ * overflows the limits of a size_t, non-zero is returned to signal failure.
+ *
+ * If the multiplication operation fails, the nature of any result stored the
+ * provided pointer is undefined, as is whether a result is stored at all.
+ *
+ * @param result
+ *     A pointer to the size_t variable that should receive the result of
+ *     multiplying the given values.
+ *
+ * @param factor_count
+ *     The number of factors to multiply together.
+ *
+ * @param factors
+ *     An array of one or more size_t values that should be multiplied
+ *     together. At least one value MUST be provided.
+ *
+ * @returns
+ *     Zero if the multiplication was successful and did not overflow the
+ *     limits of a size_t, non-zero otherwise.
+ */
+int PRIV_guac_mem_ckd_mul(size_t* result, size_t factor_count, const size_t* factors);
+
+/**
+ * Adds together each of the given values, storing the result in a size_t
+ * variable via the provided pointer. If the result of the addition overflows
+ * the limits of a size_t, non-zero is returned to signal failure.
+ *
+ * If the addition operation fails, the nature of any result stored the
```

Comment:
Looks like a missing word here: "...of any result stored the provided pointer..."

Reply:
Fixed via rebase.